src/services/firebase/hackathon.ts

```
@@ -0,0 +1,35 @@
+import { firestore } from "@/services/firebase";
+import { logEvent } from "@/services/firebase/log";
+import { doc, getDoc } from "firebase/firestore";
+
+const HACKATHONS_COLLECTION = "hackathons";
+
+export interface HackathonDates {
+	appOpenDate: string;
+	appCloseDate: string;
+}
+
+export async function getHackathonDates(year:string): Promise<HackathonDates> {
+	try {
+		const docRef = doc(firestore, HACKATHONS_COLLECTION, year);
+		const docSnap = await getDoc(docRef);
+
+		if (!docSnap.exists()) {
+			return null;
```

Can we raise an exception instead of returning `null`? Same for line 30